e2e/e2e_test.go

```diff
@@ -544,246 +559,245 @@ func TestCredit(t *testing.T) {
 		require.Equal(t, expected, resp.JSON201)
 	})
 
-	var ledgerID credit.LedgerID
-	ledgerMeta := map[string]string{
-		"test": "data",
-	}
-	t.Run("Create Ledger", func(t *testing.T) {
-		resp, err := client.CreateLedgerWithResponse(context.Background(), api.CreateLedgerJSONRequestBody{
-			Subject:  subject,
-			Metadata: ledgerMeta,
-		})
+	// FIXME: oapi-codegen is broken
```

Comment:
I assume this is required for PR

Reply:
tbh i'd be happy to hear suggestions... 
TLDR for those not familiar: `oapi-codegen` cannot resolve complex types (like `oneOf`, `anyOf` ...). we can monkeypatch-it by manually typing out the types (which i disdain) or we can give up our aspirations of using those types (that i also disdain)... thoughts?

Reply:
fixed the labour-intensive way